### Example output count file from ./Get_variations/mpileup_count.py 
* `CHR`: chromosome as in the FASTA file used for alignment
* `bp`: position
* `REF`: reference allele
* `A`: Number of A
* `G`: Number of G
* `C`: Number of C
* `T`: Number of T
* `del`: Number of deletions
* `ins`: Number of insertions
* `inserted`: allele inserted
* `ambigous`: ambigous base call
* `Depth`: depth at the site
* `N_REF`: Number of reference allele
* `_REF`: proportion of reference allele

CHR	bp	REF	A	G	C	T	del	ins	inserted	ambiguous	Depth	N_REF	%REF<br/>
GFP684	7	G	0	512	0	0	0	0	.	.	512	512	1.0<br/>
GFP684	8	C	0	2	480	0	0	0	.	.	482	480	0.995850622407<br/>
GFP684	9	T	0	0	0	558	0	0	.	.	558	558	1.0<br/>
GFP684	10	T	1	0	0	616	0	0	.	.	617	616	0.998379254457<br/>
GFP684	11	G	15	670	0	0	0	0	.	.	685	670	0.978102189781<br/>
