#!/bin/bash

# INPUT example file: CX20_n5_mapq30_sorted.bam
# OUTPUT example file: CX20_n5_mapq30_sorted_qual30.count

DIR=$1 # source code directory
REF=$2 # reference genome (same as the one used for mapping the sequencing reads)
BAM_DIR=$3 # BAM directory containing filtered bam files to be analyzed
WORK_DIR=$4 # Working directory: where to output the counts generated by samtools mpileup and mpileup_count.py
mismatch=$5 # number of mismatches allowed when mapping sequences (only for naming convention - argument can be removed)
N=$6 # Sample number
QUAL=$7 # skip bases with baseQ/BAQ smaller than INT
count_SCRIPT=$DIR/Get_variations/mpileup_count.py
DEPTH=1000000 # max per-BAM depth to avoid excessive memory usage


samtools mpileup -d $DEPTH -Q $QUAL -BAf $REF $BAM_DIR/CX${N}_n${mismatch}_mapq30_sorted.bam>$WORK_DIR/CX${N}_n${mismatch}_mapq30_sorted_qual${QUAL}.mpileup
python $count_SCRIPT $WORK_DIR/CX${N}_n${mismatch}_mapq30_sorted_qual${QUAL}.mpileup > $WORK_DIR/CX${N}_n${mismatch}_mapq30_sorted_qual${QUAL}.count
